app/actions/views/channel.js

```diff
@@ -572,6 +574,72 @@ function setLoadMorePostsVisible(visible) {
     };
 }
 
+function loadGroupData() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const actions = [];
+        const team = getCurrentTeam(state);
+        const teamID = team.id;
+        const serverVersion = state.entities.general.serverVersion;
+        const license = getLicense(state);
+        const hasLicense = license?.IsLicensed === 'true' && license?.LDAPGroups === 'true';
+
+        if (hasLicense && team && isMinimumServerVersion(serverVersion, 5, 24)) {
+            for (let i = 0; i <= MAX_RETRIES; i++) {
+                try {
+                    if (team.group_constrained) {
+                        const [getAllGroupsAssociatedToChannelsInTeam, getAllGroupsAssociatedToTeam] = await Promise.all([ //eslint-disable-line no-await-in-loop
+                            Client4.getAllGroupsAssociatedToChannelsInTeam(teamID, true),
+                            Client4.getAllGroupsAssociatedToTeam(teamID, true),
+                        ]);
+
+                        if (getAllGroupsAssociatedToChannelsInTeam.groups) {
+                            actions.push({
+                                type: GroupTypes.RECEIVED_ALL_GROUPS_ASSOCIATED_TO_CHANNELS_IN_TEAM,
+                                data: {groupsByChannelId: getAllGroupsAssociatedToChannelsInTeam.groups},
+                            });
+                        }
+
+                        if (getAllGroupsAssociatedToTeam) {
+                            actions.push({
+                                type: GroupTypes.RECEIVED_ALL_GROUPS_ASSOCIATED_TO_TEAM,
+                                data: {...getAllGroupsAssociatedToTeam, teamID},
```

Comment:
I can infer that `getAllGroupsAssociatedToTeam` has a `group` property needed by the reducer and you decided to deconstructed here but why do you  need `teamID` I don't see it being used in the reducer?

Reply:
![image](https://user-images.githubusercontent.com/17804942/86409826-a13da500-bc87-11ea-8db8-29d2970fe0af.png)

Reply:
oh ok this is on an existing reducer